src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,297 @@
+                        // put into caller
+                        return true;
+                    }
+                    else
+                    {
+                        // If there is only one argument, it has 3 cases
+                        // 1. This argument is an array (identifier)
+                        // 2. This argument is an array (array creation expression)
+                        // 3. This argument is the single element in the array. (identifier or literal)
+                        // Generate the declaration for case 2 & 3
+                        var argument = arguments[0];
+                        var typeOfElement = (IArrayTypeSymbol)parameterSymbol.Type;
+                        var typeOfArgument = semanticModel.GetTypeInfo(argument, cancellationToken).Type;
+                        return !(typeOfElement.Equals(typeOfArgument) && syntaxFacts.IsIdentifierName(argument));
+                    }
+                }
+
+                return false;
+            }
+
+            private static bool ParameterWithLiteralArgumentFilter(
+                ISyntaxFacts syntaxFacts,
+                (IParameterSymbol parameterSymbol, ImmutableArray<SyntaxNode> arguments) parameterAndArguments)
+            {
+                var (parameterSymbol, arguments) = parameterAndArguments;
+                if (!parameterSymbol.IsParams && arguments.Length == 1)
+                {
+                    return syntaxFacts.IsLiteralExpression(arguments[0]);
+                }
+
+                return false;
+            }
+
+            private static bool ParameterWithVariableDeclarationArgumentFilter(
+                ISyntaxFacts syntaxFacts,
+                (IParameterSymbol parameterSymbol, ImmutableArray<SyntaxNode> arguments) parameterAndArguments)
+            {
+                var (_, arguments) = parameterAndArguments;
+                return arguments.Length == 1 && syntaxFacts.IsDeclarationExpression(arguments[0]);
+            }
+
+            private static bool ParameterWithIdentifierArgumentFilter(
+                ISyntaxFacts syntaxFacts,
+                SemanticModel semanticModel,
+                (IParameterSymbol parameterSymbol, ImmutableArray<SyntaxNode> arguments) parameterAndArguments,
+                CancellationToken cancellationToken)
+            {
+                var (parameterSymbol, arguments) = parameterAndArguments;
+                if (arguments.Length == 1)
+                {
+                    var argument = arguments[0];
+                    var isIdentifier = syntaxFacts.IsIdentifierName(argument);
+                    if (parameterSymbol.IsParams && isIdentifier)
+                    {
+                        // If there is only one identifier argument,
+                        // it could be an identifier to an array or single element.
+                        // Only treat the array case as 'identifier'.
```

Comment:
what if it's an `int[]` passed to a `params object[]`?

Reply:
Switch to use the IOperation let these checks become unnecessary. :) So mark it as resovled.